src/main/java/at/ac/fhcampuswien/fhmdb/models/MovieAPIRequestBuilder.java

```diff
@@ -0,0 +1,79 @@
+package at.ac.fhcampuswien.fhmdb.models;
+
+import java.lang.reflect.Field;
+
+public class MovieAPIRequestBuilder {
+
+    private String base;
+
+    private String query;
+
+    private String genre;
+
+    private String releaseYear;
+
+    private String ratingFrom;
+
+    public MovieAPIRequestBuilder(String base) {
+        this.base = base;
+    }
+
+    public MovieAPIRequestBuilder query(String query) {
+        this.query = query;
+        return this;
+    }
+
+    public MovieAPIRequestBuilder genre(String genre) {
+        this.genre = genre;
+        return this;
+    }
+
+    public MovieAPIRequestBuilder releaseYear(String releaseYear) {
+        this.releaseYear = releaseYear;
+        return this;
+    }
+
+    public MovieAPIRequestBuilder ratingFrom(String ratingFrom) {
+        this.ratingFrom = ratingFrom;
+        return this;
+    }
+
+    public String build() {
+        String regex = "/\\d+$";
+        boolean isSpecificMovieRequested = base.matches(regex);
+        if (isSpecificMovieRequested) {
+            return base;
+        } else {
+            StringBuilder url = new StringBuilder(base);
+            boolean isFirstQueryParameter = true;
+            for (Field field : this.getClass().getDeclaredFields()) {
```

Comment:
I would really like to avoid using reflection, it should be the very last tool you would use when implementing a task. It breaks code abstractions, has lower performance and you loose the advantages of Java's static typing.